src/inlineSuggestions/acceptInlineSuggestion.ts

```diff
@@ -36,9 +43,28 @@ export default async function acceptInlineSuggestion(
           editor,
           prefix
         ));
+
+    const vimActive = extensions.getExtension("vscodevim.vim")?.isActive;
+    if (vimActive) {
+      await vimReturnToInsertMode(currentSuggestion);
+    }
   }
 }
 
+async function vimReturnToInsertMode(suggestion: ResultEntry) {
+  await commands.executeCommand("extension.vim_escape");
+  await commands.executeCommand("extension.vim_insert");
+  const nextCursorPosition =
+    suggestion.new_prefix.length + suggestion.new_suffix.length;
+  setImmediate(() => {
```

Comment:
What happens without this is that the cursor jumps to the start of the completion after this callback is finished. Something moves the cursor again before the tick ends, and as far as I can tell it's not Tabnine because the cursor movement listener isn't triggered when it happens. Moving the cursor on the next tick seems to address that and work well.

Reply:
Turns out that we need to use the `vim` move cursor commands to achieve the desired result without manipulating the event order further. Hurray!